### PR TITLE
Template pull requests

### PR DESCRIPTION
**Pull request by briangrahamww**
Adding a template for pull requests.

## Description
Copying this over from our private repo:

I have taken some inspiration from one of my favorite (and well managed!) repo's (Spacy) and added a PR template.  The template has a markdown checklist that you should check off for each PR.

### Types of change
Feature enhancement

## Checklist

- [x] I squashed my commits to a reasonable number of descriptive commits.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by kzecchini** [comment]
Nice - I really like this. I also think this might be a good time to add a `CHANGELOG.md` file and have a guideline on adding to the changelog as well.

Also we don't need to squash commits before the PR necessarily, we can squash on merge as well which I think is a viable option.

**Comment by kzecchini** [comment]
@briangrahamww I made a PR #60 which has a default auto generated changelog file. Not sure if we want to have these changes done manually for each PR (reminder in the checklist), or to autogenerate the changes on build. Let me know what you think is best.

**Comment by briangrahamww** [comment]
> @briangrahamww I made a PR #60 which has a default auto generated changelog file. Not sure if we want to have these changes done manually for each PR (reminder in the checklist), or to autogenerate the changes on build. Let me know what you think is best.

Nice! I'll take a look at that. Catching up on two weeks of git activity now that I'm back...

Also, this needs to be re-approved. Had to merge in master since this became out of date.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+or new feature, or a change to the documentation? -->
+
+## Checklist
+<!--- Before you submit the PR, go over this checklist and make sure you can
+tick off all the boxes. [] -> [x] -->
+- [ ] I squashed my commits to a reasonable number of descriptive commits.
+- [ ] I ran the tests, and all new and existing tests passed.
+- [ ] My changes don't require a change to the documentation, or if they do, I've added all required information.
+
+<!--- Note: I copied this from the Spacy repo, and modified sightly for our rules. So credit to them!-->
```
Comment (kzecchini) [inline .github/PULL_REQUEST_TEMPLATE.md]: This is a great shoutout - but not sure it is necessary.

**Pull request merged**